src/Functions/FunctionsBitmap.h

```diff
@@ -282,18 +284,16 @@ class FunctionBitmapSubset : public IFunction
                 "First argument for function " + getName() + " must be a bitmap but it has type " + arguments[0]->getName() + ".",
                 ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
 
-        const auto * arg_type1 = typeid_cast<const DataTypeNumber<UInt32> *>(arguments[1].get());
-        if (!(arg_type1))
-            throw Exception(
-                "Second argument for function " + getName() + " must be UInt32 but it has type " + arguments[1]->getName() + ".",
-                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
-
-        const auto * arg_type2 = typeid_cast<const DataTypeNumber<UInt32> *>(arguments[1].get());
-        if (!(arg_type2))
-            throw Exception(
-                "Third argument for function " + getName() + " must be UInt32 but it has type " + arguments[2]->getName() + ".",
-                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
-
+        for (size_t i = 1; i < 3; ++i)
+        {
+            WhichDataType which(arguments[i].get());
+            if (!(which.isUInt8() || which.isUInt16() || which.isUInt32() || which.isUInt64()))
+            {
+                throw Exception(
+                    "The second or third argument for function " + getName() + " must be one of [UInt8, UInt16, UInt32, UInt64] but it has type " + arguments[1]->getName() + ".",
```

Comment:
`The second or third argument must be one of ...` sounds a bit weird (like one of them must be and the second is not). I think it would be better to write an accurate argument number or make it `The second and third arguments ... but one of them has type ...`

Reply:
Right